client/src/Components/CommonComponents/AutoCompleteTags/index.jsx

```diff
@@ -18,17 +18,21 @@ export default class AutoCompleteTags extends Component {
   componentDidMount() {
     // git the data form the back for skills and offer_type and setSatet for the skills and offer type
     // const selected = options.filter(item => _.find(fetched, id => id === item.id))
-    const { data, type } = this.props;
-    this.setState(() => {
-      let options;
-      if (type === 'skill') {
-        options = skills;
-      } else if (type === 'offer_type') {
-        options = offerType;
+    const { type } = this.props;
+    this.setState(
+      () => {
+        let options;
+        if (type === 'skill') {
+          options = skills;
+        } else if (type === 'offer_type') {
+          options = offerType;
+        }
+        return { options };
+      },
+      () => {
+        this.setState({ selectedTags: this.props.data });
```

Comment:
@Fatmasiam I think you could shorten this a little bit, if you wanted, into one call to setState instead of two.  That might result in better performance, since it would always cause one component render() to happen instead of sometimes two renders.
Something like this:
```
const { type, data } = this.props
let options;
if (type === 'skill') {
  options = skills;
} else if (type === 'offer_type') {
  options = offerType;
}
this.setState({
  options,
  selectedTags: data
})
```
You can see that it's just one setState happening now, and also that it's just passing a simple object to setState instead of passing it an updater function.

Reply:
this case didn't work with me when I use the AutoCompleteTage twice on the same page,
the data props just file to the second AutoCompleteTage like in the Home Component #79 
because of the initial render so I make it like this

Reply:
Oh, then I wonder if you need to actually make a copy of this.props.data when you call setState so that two components can't mess with each others' props:
so change this:
`selectedTags: data` to this:
`selectedTags: { ...data }` (or [...data] if props.data is an array)
Maybe that will fix the issue.  But I don't want to take too much of your time, so if your code works then stick with it :)